i6_models/assemblies/util.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import MISSING, dataclass
+from typing import Any, List, Optional
+
+
+@dataclass
+class ModelConfiguration:
+    """base dataclass that supported fetching attributes"""
+
+    _name: Optional[str] = None
+
+    @staticmethod
+    def name():
+        return None
+
+    def _get_all_attributes(self) -> List[str]:
+        return [k for k in self.__dataclass_fields__.keys()]
+
+    def _get_name(self, attribute_name: str) -> str:
+        return self.__dataclass_fields__[attribute_name].name
+
+    def _get_default(self, attribute_name: str) -> Any:
+        if hasattr(self, attribute_name):
+            return getattr(self, attribute_name)
+
+        f = self.__dataclass_fields__[attribute_name]
+        if f.default_factory is not MISSING:
+            return f.default_factory()
+        return f.default
+
+    def _get_type(self, attribute_name: str) -> Any:
+        return self.__dataclass_fields__[attribute_name].type
+
+    @classmethod
+    def from_namespace(cls, args):
```

Comment:
do we really want to silently drop all config params that do not match? This seems error prone to me.

Why do we even need such a config class converter?

Reply:
So the usecase I found to not drop this was that you could have one big custom Config to use for your model which then initializes each subconfig you later need for the model.
So e.g. `TransformerConfig` can then initialize `TransformerEncoder` and `TransformerDecoder` etc. 
But if this makes sense or should be dropped is def. up for discussion.